### DISCUSSION TITLE
How to use C libraries?

### DISCUSSION THREAD

**Discussion by ghost** [objeck/objeck-lang#82] [General]
Like the title. Is it possible?

**Comment by objeck** [comment]
If I understand the question correctly. Yes, the system supports calls into C-shared libraries.

**Comment by objeck** [comment]
Here's an example:

The `CallFunction` function takes the name of the C-function to be called. The `VMContext `object and `APITools_XXX` functions pass data to and from the runtime.  The C-API gives programmers direct access to Objeck arrays and object instances, so care has to be taken not to corrupt them.

---
https://github.com/objeck/objeck-lang/blob/master/core/compiler/lib_src/odbc.obs#L32
https://github.com/objeck/objeck-lang/blob/master/core/lib/odbc/odbc.cpp#L71
https://github.com/objeck/objeck-lang/blob/master/core/vm/lib_api.h
